### PR TITLE
Added URL params logic to switch tabs based on URL

### PR DESCRIPTION
**Pull request by radekkaluzik**
Added URL params logic to switch tabs based on URL. Accepted params `bundle`, `activeTab `
example: `http://foo?bundle=openshift&activeTab=configuration`

[RHCLOUD-29374](https://issues.redhat.com/browse/RHCLOUD-29374)

---

### Checklist ☑️
- [x] PR only fixes one issue or story 
- [x] Change reviewed for extraneous code 
- [x] UI best practices adhered to 
- [x] Commits squashed and meaningfully named 
- [x] All PR checks pass locally (build, lint, test, E2E)

##
- [ ] _(Optional) QE: Needs QE attention (OUIA changed, perceived impact to tests, no test coverage)_
- [ ] _(Optional) QE: Has been mentioned_
- [ ] _(Optional) UX: Needs UX attention (end user UX modified, missing designs)_
- [ ] _(Optional) UX: Has been mentioned_
##

### CODE REVIEW

File: src/pages/Notifications/List/BundlePage.tsx [hunk 1]
```diff
@@ -23,22 +23,52 @@ import { linkTo } from '../../../Routes';
...
   bundleTabs: Facet[];
   applications: Array<Facet>;
 }
 
+enum BundleTabIndex {
+  Rhel = 0,
+  Console = 1,
+  OpenShift = 2,
+}
+
```
Comment (fhlavac) [inline src/pages/Notifications/List/BundlePage.tsx]: we could probably map the bundle names directly to the indexes and avoid the switch

File: src/pages/Notifications/List/BundlePageBehaviorGroupContent.tsx [hunk 2]
```diff
@@ -29,18 +29,32 @@ enum TabIndex {
   BehaviorGroups = 1,
 }
 
+const useActiveTab = (): TabIndex => {
+  const location = useLocation();
+  const queryParams = new URLSearchParams(location.search);
+  const activeTabParam = queryParams.get('activeTab');
+
+  const activeTab: TabIndex = (() => {
+    switch (activeTabParam) {
```
Comment (fhlavac) [inline src/pages/Notifications/List/BundlePageBehaviorGroupContent.tsx]: ^^

File: src/pages/Notifications/List/BundlePageBehaviorGroupContent.tsx [hunk 3]
```diff
@@ -24,23 +24,32 @@ interface BundlePageBehaviorGroupContentProps {
...
+  useEffect(() => {
+    const queryParams = new URLSearchParams(location.search);
+    const activeTabParam = queryParams.get('tab') ?? 'configuration';
+    const activeTabKey = tabMapping[activeTabParam] ?? 0;
+    console.log(
+      'activeTabParam: ',
+      activeTabParam,
+      'activeTabKey: ',
+      activeTabKey
+    );
```
Comment (fhlavac) [inline src/pages/Notifications/List/BundlePageBehaviorGroupContent.tsx]:
```suggestion
```

**Comment by fhlavac** [comment]
When adding `tab=behaviorGroups` to the URL, "Configuration" seems to be still active.

Also, when loading the page for the first time, no URL params identifying active tabs are present. When you change the tabs, the URL params do not update. The current logic is probably enough to open the appropriate tab when pointing here from the drawer, but does not cover what we need for routing changes [RHCLOUD-28284](https://issues.redhat.com/browse/RHCLOUD-28284). Not sure what you've agreed on with Cody.

File: src/pages/Notifications/List/BundlePage.tsx (hunk 1 above)
Comment (radekkaluzik) [reply src/pages/Notifications/List/BundlePage.tsx]: fixed

File: src/pages/Notifications/List/BundlePageBehaviorGroupContent.tsx (hunk 2 above)
Comment (radekkaluzik) [reply src/pages/Notifications/List/BundlePageBehaviorGroupContent.tsx]: Fixed

File: src/pages/Notifications/List/BundlePage.tsx [hunk 4]
```diff
@@ -23,17 +23,33 @@ import { linkTo } from '../../../Routes';
...
   const { updateDocumentTitle } = useChrome();
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const { bundle } = useMemo(() => {
+    const searchParams = new URLSearchParams(location.search);
+    return {
+      bundle: searchParams.get('bundle') ?? 'rhel',
+    };
+  }, [location.search]);
```
Comment (fhlavac) [inline src/pages/Notifications/List/BundlePage.tsx]:
```suggestion
const bundle = useMemo(
    () => new URLSearchParams(location.search).get('bundle') ?? 'rhel',
    [location.search]
  );
```

File: src/pages/Notifications/List/BundlePage.tsx [hunk 5]
```diff
@@ -86,17 +101,13 @@ export const NotificationListBundlePage: React.FunctionComponent<
     </Main>
   );
 
+  // TO DO
```
Comment (fhlavac) [inline src/pages/Notifications/List/BundlePage.tsx]:
```suggestion
```

File: src/pages/Notifications/List/BundlePage.tsx [hunk 6]
```diff
@@ -86,17 +101,13 @@ export const NotificationListBundlePage: React.FunctionComponent<
...
-      query(props.bundleTabs[activeTabKey].name);
+      query(props.bundleTabs[bundleMapping[bundle]].name);
     }
-  }, [
-    activeTabKey,
-    getApplications.query,
-    props.bundleTabs,
-    notificationsOverhaul,
-  ]);
+  }, [bundle]);
```
Comment (fhlavac) [inline src/pages/Notifications/List/BundlePage.tsx:119]:
```suggestion
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [bundle]);
```

File: src/pages/Notifications/List/BundlePage.tsx [hunk 7]
```diff
@@ -126,8 +133,19 @@ export const NotificationListBundlePage: React.FunctionComponent<
...
+                newSearchParams.set(
+                  'bundle',
+                  Object.entries(bundleMapping).find(
+                    (entry) => entry[1] === tabIndex
+                  )?.[0] ?? 'rhel'
+                );
+                navigate(`${location.pathname}?${newSearchParams.toString()}`, {
+                  replace: true,
+                });
+              }}
```
Comment (fhlavac) [inline src/pages/Notifications/List/BundlePage.tsx]:
```suggestion
onSelect={(event, tabIndex) => {
                const newSearchParams = new URLSearchParams(location.search);
                const selectedBundle = Object.keys(bundleMapping).find(
                  (key) => bundleMapping[key] === tabIndex
                );
                newSearchParams.set('bundle', selectedBundle ?? 'rhel');
                navigate(`${location.pathname}?${newSearchParams.toString()}`, {
                  replace: true,
                });
              }}
```

File: src/pages/Notifications/List/BundlePageBehaviorGroupContent.tsx [hunk 8]
```diff
@@ -24,27 +23,31 @@ interface BundlePageBehaviorGroupContentProps {
...
-    queryParams.get('activeTab') === 'behaviorGroups'
-      ? TabIndex.BehaviorGroups
-      : TabIndex.Configuration;
+
+  const { tab } = useMemo(() => {
+    const searchParams = new URLSearchParams(location.search);
+    return {
+      tab: searchParams.get('tab') ?? 'configuration',
+    };
+  }, [location.search]);
```
Comment (fhlavac) [inline src/pages/Notifications/List/BundlePageBehaviorGroupContent.tsx]:
```suggestion
  const tab = useMemo(() => new URLSearchParams(location.search).get('tab') ?? 'configuration', [location.search]);
```

File: src/pages/Notifications/List/BundlePageBehaviorGroupContent.tsx [hunk 9]
```diff
@@ -106,32 +109,43 @@ export const BundlePageBehaviorGroupContent: React.FunctionComponent<
...
+        const newSearchParams = new URLSearchParams(location.search);
+        newSearchParams.set(
+          'tab',
+          Object.entries(tabMapping).find(
+            (entry) => entry[1] === tabIndex
+          )?.[0] ?? 'configuration'
+        );
+        navigate(`${location.pathname}?${newSearchParams.toString()}`, {
+          replace: true,
+        });
```
Comment (fhlavac) [inline src/pages/Notifications/List/BundlePageBehaviorGroupContent.tsx]:
```suggestion
const newSearchParams = new URLSearchParams(location.search);
        const selectedTab =
          Object.keys(tabMapping).find((key) => tabMapping[key] === tabIndex) ??
          'configuration';
        newSearchParams.set('tab', selectedTab);
        navigate(`${location.pathname}?${newSearchParams.toString()}`, {
          replace: true,
        });
```

**Comment by fhlavac** [comment]
@radekkaluzik looks great, just some cleanup notes 🙂

File: src/pages/Notifications/List/BundlePage.tsx (hunk 4 above)
Comment (radekkaluzik) [reply src/pages/Notifications/List/BundlePage.tsx]: fixed

File: src/pages/Notifications/List/BundlePage.tsx (hunk 5 above)
Comment (radekkaluzik) [reply src/pages/Notifications/List/BundlePage.tsx]: Fixed

File: src/pages/Notifications/List/BundlePage.tsx (hunk 6 above)
Comment (radekkaluzik) [reply src/pages/Notifications/List/BundlePage.tsx:119]: Fixed

File: src/pages/Notifications/List/BundlePage.tsx (hunk 7 above)
Comment (radekkaluzik) [reply src/pages/Notifications/List/BundlePage.tsx]: Fixed

File: src/pages/Notifications/List/BundlePageBehaviorGroupContent.tsx (hunk 8 above)
Comment (radekkaluzik) [reply src/pages/Notifications/List/BundlePageBehaviorGroupContent.tsx]: Fixed

File: src/pages/Notifications/List/BundlePageBehaviorGroupContent.tsx (hunk 9 above)
Comment (radekkaluzik) [reply src/pages/Notifications/List/BundlePageBehaviorGroupContent.tsx]: Fixed

**Comment by fhlavac** [comment]
@radekkaluzik looks great, the only thing I am noticing is that the URL params are not present on an initial render

**Comment by radekkaluzik** [comment]
@fhlavac fixed - params on initial page render

**Review by fhlavac — APPROVED** [review]

**Pull request merged**